src/components/Form/SignInModal/SignIn/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { Button, ErrorMessage, Input } from '@/components'
+import { useFormik } from 'formik'
+import { useAuth } from '@/context'
+import {
+  Title,
+  Text,
+  ChangeFormBtn,
+  Wrapper,
+} from '@/components/Form/SignInModal/styles'
+import { isSome, none, fromNullable, chain, getLeft } from 'fp-ts/Option'
+import * as Yup from 'yup'
+import { pipe } from 'fp-ts/lib/function'
+
+type SignInProps = {
+  onSwitchFormClick: (state: boolean) => void
+}
+
+export type SignInValues = {
+  email: string
+  password: string
+}
+
+const signInSchema = Yup.object({
+  email: Yup.string()
+    .email('E-mail is not valid')
+    .required('Email field is required')
+    .lowercase(),
+  password: Yup.string()
+    .required('Password field is required')
+    .min(8, 'Minimum 8 characters required!')
+    .max(64, 'Maximum 64 characters required!'),
+})
+
+const SignIn = ({ onSwitchFormClick }: SignInProps) => {
+  const { signIn, isLoading, data, error: genericError } = useAuth()
+
+  const initialValues: SignInValues = {
+    email: '',
+    password: '',
+  }
+
+  const formik = useFormik({
+    initialValues: initialValues,
+    validationSchema: signInSchema,
+    onSubmit: (values: SignInValues) => handleSubmit(values),
+  })
+
+  const handleSubmit = (values: SignInValues) => {
+    const newSignInValues = {
+      user: values,
+    }
+    signIn(newSignInValues)
+  }
+
+  const dataError = pipe(
```

Comment:
`**` Não sei, não to curtindo muito esse código aqui. Vamos falar em call sobre ele.
Vou anotar com ** tudo que formos falar em call.